### PR TITLE
Added missing `test_tokenization_led`

### PR DESCRIPTION
**Pull request by IMvision12**
# What does this PR do?

Added missing `test_tokenization_led`, was similar to Bart tokenizer made some changes by testing it in local environment
@sgugger

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for adding this! LGTM, can you also have a quick look @ydshieh ?

**Review by ydshieh — APPROVED** [review]
Thank you @IMvision12 for adding this missing test 💯 !

It LGTM. I have a suggestion though: as LED has `global_attention_mask`, and there is a slightly different behavior in padding method for `LEDTokenizer(Fast)` regarding it (see below), I think it is a good idea to add a test method for this.

The test could be: having 2 texts, encoding them without padding (where the length will be different), and send `encoded_inputs` together with `global_attention_mask` (we have to create it) to `pad` method.

Let me know if you need more context, thanks!

https://github.com/huggingface/transformers/blob/5764efe544de909e93bfde16489b5a0975fce1c2/src/transformers/models/led/tokenization_led.py#L444-L462

**Comment by IMvision12** [comment]
@ydshieh Can you give some more points of what exactly is to be done?

As per the points given by you, I need to first create 2 texts let's say `A long paragraph for summarization.` and `Another paragraph for`, and then encode them as `tokenizer.encode_plus("Another paragraph for", padding=False)` passing padding as False so that it doesn't apply padding to text, and then we have to create a list of `global_attention_mask` let's say [0,0,0,0,0], doing this for both the text and then pass encoded_inputs along with `global_attention_mask` to the `tokenizer._pad()`

**Comment by ydshieh** [comment]
@IMvision12 Yes, that's the idea :-). Only at the end, you can do `tokenizer.pad()` instead -> it will call `_pad` internally.

**Comment by IMvision12** [comment]
@ydshieh Also what I really need to check in `assertEqual`?

**Comment by ydshieh** [comment]
We need to check the outputs after padding contains the key `global_attention_mask` and its value is the same as the expected one, which is the `global_attention_mask` being padded. You will either have to take a quick look in ` _pad`  or at least run one example to get a better idea (which should be easy enough) what it does :-)

**Comment by IMvision12** [comment]
@ydshieh can you take a quick look at this function
Is this expected to be done?

```
    def test_global_attention(self):
        text = ["A long paragraph for summarization.", "Another paragraph for summarization."]
        tokenizer = self.default_tokenizer_fast()
        
        input_1 = tokenizer.encode_plus(text[0], padding=False)
        input_1['global_attention_mask'] = [0,0,0,0,0]
        outputs_1 = tokenizer.pad(input_1)
        self.assertEqual(outputs_1['global_attention_mask'],[0, 0, 0, 0, 0, -1, -1, -1, -1])

        input_2 = tokenizer.encode_plus(text[1], padding=False)
        input_2['global_attention_mask'] = [0,0,0,0]
        outputs_2 = tokenizer.pad(input_2)
        self.assertEqual(outputs_2['global_attention_mask'],[0, 0, 0, 0, -1])

```

**Comment by ydshieh** [comment]
@IMvision12 

The idea is to encode the 2 texts together without padding, and send the encoded outputs with `global_attention_mask` (not padded neither) to `.pad`.

You code above pads each sequence, which won't have any padding. The padding only happens with multiple sequences where the length are different.

**Comment by IMvision12** [comment]
@ydshieh sorry for pinging you so many times

```
def test_global_attention_mask(self):
        text = ["A long paragraph.", "Hi I am using huggingface transformers"]
        tokenizer = self.default_tokenizer_fast()
        
        inputs = tokenizer.encode_plus(text, padding=False)
        inputs['global_attention_mask'] = [0,0,0,0,0,0,0,0]
        outputs = tokenizer.pad(inputs)
        self.assertEqual(outputs['global_attention_mask'],[0, 0, 0, 0, 0, 0, 0, 0, -1, -1, -1, -1, -1, -1, -1, -1])
```

File: tests/models/led/test_tokenization_led.py [hunk 1]
```diff
@@ -171,3 +171,16 @@ def test_embeded_special_tokens(self):
                 self.assertSequenceEqual(
                     tokens_r_str, ["<s>", "A", ",", "<mask>", "ĠAllen", "N", "LP", "Ġsentence", ".", "</s>"]
                 )
+
+    @require_torch
+    def test_global_attention_mask(self):
+        for tokenizer, pretrained_name, kwargs in self.tokenizers_list:
+            with self.subTest(f"{tokenizer.__class__.__name__} ({pretrained_name})"):
+                tokenizer_r = self.rust_tokenizer_class.from_pretrained(pretrained_name, **kwargs)
+                src_text = ["A long paragraph.", "Hi I am using huggingface transformers"]
```
Comment (ydshieh) [inline tests/models/led/test_tokenization_led.py:180]: Let's use `A short text` maybe? It's the shorter one of the two texts :-)
Don't forget to update the necessary mask and expected outputs

File: tests/models/led/test_tokenization_led.py [hunk 2]
```diff
@@ -171,3 +171,16 @@ def test_embeded_special_tokens(self):
...
+                expected_global_attention_mask = [[0, 0, 0, 0, 0, 0, -1, -1, -1, -1], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]]
+
+                inputs = tokenizer_r(src_text, padding=False)
+                inputs["global_attention_mask"] = [[0] * len(y) for y in inputs["input_ids"]]
+                outputs = tokenizer_r.pad(inputs)
+                self.assertSequenceEqual(outputs["global_attention_mask"], expected_global_attention_mask)
```
Comment (ydshieh) [inline tests/models/led/test_tokenization_led.py:186]: Let's also check the slow tokenizer too 🙏

**Review by ydshieh — APPROVED** [review]
Hi @IMvision12 Thank you. Leave final comment and we are good to merge 💯

**Comment by IMvision12** [comment]
I am not sure `why tests_pipelines_tf` are failing

**Comment by ydshieh** [comment]
No need to worry about the TF pipeline test. I will take a look - it's probably irrelevant to this PR.

**Comment by ydshieh** [comment]
Could you update your local main branch , and rebase your working branch on local `main`?

**Comment by IMvision12** [comment]
@ydshieh Done!

**Review by ydshieh — APPROVED** [review]
Thank you for the contribution 🚀 @IMvision12 !

**Comment by IMvision12** [comment]
@ydshieh Thanks for a concise explanation of `global_attention_mask` and guidance!!
